django/db/models/query.py

```diff
@@ -1147,7 +1147,8 @@ def in_bulk(self, id_list=None, *, field_name="pk"):
             if not id_list:
                 return {}
             filter_key = "{}__in".format(field_name)
-            batch_size = connections[self.db].features.max_query_params
+            num_fields = len(opts.pk_fields) if field_name == "pk" else 1
+            batch_size = connections[self.db].features.max_query_params // num_fields
```

Comment:
This was [uncovered before](https://djangoci.com/view/%C2%ADCoverage/job/django-coverage/HTML_20Coverage_20Report/z_d81526da7cfdb7e4_query_py.html#t1155)